frontend/src/features/mapSideBar/SidebarContents/BuildingForm.tsx

```diff
@@ -339,8 +339,11 @@ const BuidingForm: React.FC<IParentBuildingForm> = ({
   const initialValues = {
     activeStep: 0,
     activeTab: 0,
-    data: { ...defaultBuildingValues, agencyId: keycloak.agencyId, ...rest.initialValues },
+    data: { ...defaultBuildingValues, ...rest.initialValues },
   };
+  initialValues.data.agencyId = initialValues.data.agencyId
```

Comment:
Thanks @jdtoombs, for the update to land, think its required for buildings as well.